src/collector/HardwareCollector.spec.js

```diff
@@ -0,0 +1,81 @@
+/* eslint-disable no-restricted-properties */
+/* eslint-disable no-underscore-dangle */
+
+import HardwareCollector from './HardwareCollector';
+
+describe('HardwareCollector', () => {
+  const originalEvent = {
+    id: '11bf5b37-e0b8-42e0-8dcf-dc8c4aefc000',
+    clientId: '11bf5b37-e0b8-42e0-8dcf-dc8c4aefc001',
+    eventType: 'state',
+    eventAction: 'load',
+    eventSource: 'products/shoes/1',
+    eventCategory: 'products/shoes',
+    eventStart: 123,
+    eventEnd: 456,
+  };
+
+  const expectedEvent = {
+    ...originalEvent,
+    manufacturer: 'Samsung',
+    model: 'SCH-I535',
+    screenResolutionWidth: 0, // this should be modified if we are using Karma
```

Comment:
@robertdumitrescu MAINTAINABILITY - Regarding your comment, we may see benefit from this being any positive number.